### PR TITLE
[SPARK-8601][ML] Add an option to disable standardization for linear regression

### PR DESCRIPTION
**Pull request by holdenk**
All compressed sensing applications, and some of the regression use-cases will have better result by turning the feature scaling off. However, if we implement this naively by training the dataset without doing any standardization, the rate of convergency will not be good. This can be implemented by still standardizing the training dataset but we penalize each component differently to get effectively the same objective function but a better numerical problem. As a result, for those columns with high variances, they will be penalized less, and vice versa. Without this, since all the features are standardized, so they will be penalized the same.

In R, there is an option for this.
`standardize`  
Logical flag for x variable standardization, prior to fitting the model sequence. The coefficients are always returned on the original scale. Default is standardize=TRUE. If variables are in the same units already, you might not wish to standardize. See details below for y standardization with family="gaussian".

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala [hunk 1]
```diff
@@ -82,6 +82,15 @@ class LinearRegression(override val uid: String)
   setDefault(fitIntercept -> true)
 
   /**
+   * Set to enable scaling (standardization).
+   * Default is true.
+   * @group setParam
+   */
+  def setStandardization(value: Boolean): this.type = set(standardization, value)
+  setDefault(standardization -> true)
+
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala]: extra line

**Comment by dbtsai** [comment]
You don't cover all the test cases including with/without intercept. Also, for regParam = 0, they should converge to the same solution.

**Comment by holdenk** [comment]
@dbtsai I've extended the test coverage.

**Comment by dbtsai** [comment]
@holdenk Cool. I'll work on this tonight. Thanks.

**Comment by holdenk** [comment]
jenkins, retest this please

**Comment by holdenk** [comment]
jenkins, retest this please

**Comment by dbtsai** [comment]
@holdenk can you merge master? Thanks.

**Pull request closed without merging**